src/mesh/checkpoint_io.C

```diff
@@ -130,12 +179,88 @@ CheckpointIO::~CheckpointIO ()
 {
 }
 
-std::string extension(const std::string & s)
+processor_id_type CheckpointIO::selectSplitConfig(const std::string & input_name, header_id_type & data_size)
 {
-  auto pos = s.rfind(".");
-  if (pos == std::string::npos)
-    return "";
-  return s.substr(pos, s.size() - pos);
+  std::string header_name;
+
+  // We'll read a header file from processor 0 and broadcast.
+  if (this->processor_id() == 0)
+    {
+      header_name = headerFile(input_name, _my_n_processors);
+
+      {
+        // look for header+splits with nprocs equal to _my_n_processors
+        std::ifstream in (header_name.c_str());
+        if (!in.good())
+          {
+            // otherwise fall back to a serial/single-split mesh
+            header_name = headerFile(input_name, 1);
+            std::ifstream in (header_name.c_str());
+            if (!in.good())
+              {
+                libmesh_error_msg("ERROR: cannot locate header file for input '" << input_name << "'");
+              }
+          }
+      }
+
+      Xdr io (header_name, this->binary() ? DECODE : READ);
+
+      // read the version, but don't care about it
+      std::string input_version;
+      io.data(input_version);
+
+      // read the data type
+      io.data (data_size);
+    }
+
+  this->comm().broadcast(data_size);
+  this->comm().broadcast(header_name);
+
+  // How many per-processor files are here?
+  largest_id_type input_n_procs;
+
+  switch (data_size) {
+  case 2:
+    input_n_procs = this->read_header<uint16_t>(header_name);
+    break;
+  case 4:
+    input_n_procs = this->read_header<uint32_t>(header_name);
+    break;
+  case 8:
+    input_n_procs = this->read_header<uint64_t>(header_name);
+    break;
+  default:
+    libmesh_error();
+  }
+
+  if (!input_n_procs)
+    input_n_procs = 1;
+  return input_n_procs;
+}
+
+void CheckpointIO::cleanup(const std::string & input_name, processor_id_type n_procs)
+{
+  auto header = headerFile(input_name, n_procs);
+  auto ret = remove(header.c_str());
```

Comment:
Please `#include <cstdio>` and use `std::remove()` for this, just so it's really obvious what function you're calling.